src/ucp/proto/proto_common.c

```diff
@@ -449,6 +449,34 @@ ucp_proto_common_get_lane_perf(const ucp_proto_common_init_params_t *params,
     return status;
 }
 
+static inline ucp_lane_index_t
+ucp_proto_common_lanes_iter(const ucp_ep_config_key_t *ep_config_key,
+                            ucp_lane_map_t lane_map, ucp_lane_type_t lane_type,
+                            ucp_lane_index_t start, ucp_lane_index_t *lane)
+{
+    if (start >= UCP_MAX_LANES) {
+        return UCP_MAX_LANES;
+    }
+
+    if (lane_type == UCP_LANE_TYPE_RMA_BW) {
+        for (; start < ep_config_key->num_lanes; ++start) {
+            *lane = ep_config_key->rma_bw_lanes[start];
+            if ((*lane == UCP_NULL_LANE) || (lane_map & UCS_BIT(*lane))) {
+                break;
+            }
+        }
```

Comment:
maybe we could keep the code in ucp_proto_common_find_lanes that goes over the lane map, and just adjust the lane map in case of UCP_LANE_TYPE_RMA_BW to include the lanes in ep_config_key->rma_bw_lanes?

Reply:
pls add TODO that we want to remove this function and sort lanes according to actual expected BW

Reply:
done